django/db/models/functions/json.py

```diff
@@ -5,6 +5,66 @@
 from django.db.models.functions import Cast
 
 
+class JSONArray(Func):
+    function = "JSON_ARRAY"
+    output_field = JSONField()
+
+    def as_sql(self, compiler, connection, **extra_context):
```

Comment:
```suggestion
    def as_sql(self, compiler, connection, **extra_context):
            if not connection.features.supports_json_field:
                raise NotSupportedError(
                    "JSONFields are not supported on this database backend."
                )
```
Been reviewing https://github.com/django/django/pull/18351/ which has a top level check for json field support - I think this might want the same?